### PR TITLE
Bonnie 410 update fixtures tool

### PR DESCRIPTION
**Pull request by alexanderelliott121**
Updates the fixture generation tool to create frontend fixtures, updates backend fixture generation signature to be more streamlined/generalized.

### CODE REVIEW

File: lib/tasks/generate_fixtures.rake [hunk 1]
```diff
@@ -1,64 +1,106 @@
...
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
+    # patients: which patients should be exported.
+    #   values: each element should have three fields, firstname, lastname, displayname
+    #   ex: [[first1, last1, patient1], [first2, last2, patient2]]
+    ###
+    task :generate_frontend_fixture, [:cms_hqmf, :path, :test_name, :user_email, :measure_id] => [:environment] do |t, args|
+      user = User.find_by email: args[:user_email]
```
Comment (mayerm94) [inline lib/tasks/generate_fixtures.rake]: Should probably end if user is not found

File: lib/tasks/generate_fixtures.rake [hunk 2]
```diff
@@ -1,64 +1,106 @@
...
+      if (args[:cms_hqmf].downcase  == 'cms')  
+        measure = user.measures.find_by(cms_id: args[:measure_id])
+      elsif (args[:cms_hqmf] == 'hqmf')
+        measure = user.measures.find_by(hqmf_id: args[:measure_id])
+      else
+      end
```
Comment (mayerm94) [inline lib/tasks/generate_fixtures.rake]: Add content to else for error check

File: lib/tasks/generate_fixtures.rake [hunk 3]
```diff
@@ -1,64 +1,106 @@
...
+      
+      oid_to_vs_map = {}
+      
+      value_sets = measure.value_sets.each do |vs|
+        oid_to_vs_map[vs.oid] = vs
+      end
+      value_sets_file = File.join(fixtures_path, 'measure_data', args[:path], cms_dir, 'value_sets.json')
+      create_fixture_file(value_sets_file, JSON.pretty_generate(JSON.parse(oid_to_vs_map.to_json)))
+    
+      us_filename = "#{args[:test_name]}.json"
```
Comment (mayerm94) [inline lib/tasks/generate_fixtures.rake]: Why is the upload_summaries filename named after an argument but the archived_measures, value_sets and measures are named after what they contain, ie: measures.json, archived_measures.json and value_sets.json

File: lib/tasks/generate_fixtures.rake [hunk 4]
```diff
@@ -1,64 +1,106 @@
...
-        test_name = "#{tmp_email}_#{args[:cms_id]}_#{args[:patient_first_name]}_#{args[:patient_last_name]}"
-        test_name["\."] = "_"
-      end
-      
+    task :generate_fixtures, [:cms_hqmf, :path, :user_email, :measure_id] => [:environment] do |t, args|
       user = User.find_by email: args[:user_email]
-      if (args[:use_hqmf] == true)
-        measure = user.measures.find_by(hqmf_id: args[:cms_id])
+      if (args[:cms_hqmf].downcase  == 'cms')  
+        measure = user.measures.find_by(cms_id: args[:measure_id])
```
Comment (mayerm94) [inline lib/tasks/generate_fixtures.rake]: Getting the user and determining if the cms_hqmf arg is a cms_id or hqmf_id seems to be repeated code between rake tasks. Perhaps it could be moved to it's own function

**Review by mayerm94 — CHANGES_REQUESTED** [review]

File: lib/tasks/generate_fixtures.rake (hunk 3 above)
Comment (alexanderelliott121) [reply lib/tasks/generate_fixtures.rake]: This is a weird one.
At the moment, the infrastructure we have looks upload_summaries in a specific way: it expects three files called initialLoad, update1 and update2.
I'm not adverse to changing this or updating the load tool, but I ultimately decided not to change the existing structures in this branch.

File: lib/tasks/generate_fixtures.rake (hunk 1 above)
Comment (alexanderelliott121) [reply lib/tasks/generate_fixtures.rake]: It does, mongoid error.

**Review by mayerm94 — APPROVED** [review]

File: lib/tasks/generate_fixtures.rake [hunk 5]
```diff
@@ -1,64 +1,109 @@
...
-      end
-      
-      user = User.find_by email: args[:user_email]
-      if (args[:use_hqmf] == true)
-        measure = user.measures.find_by(hqmf_id: args[:cms_id])
+
+    def get_measure(user, cms_hqmf, measure_id)
+      if (cms_hqmf.downcase  == 'cms')  
+        measure = user.measures.find_by(cms_id: measure_id)
+      elsif (cms_hqmf == 'hqmf')
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: should this also be `downcase`?

Comment (edeyoung) [reply lib/tasks/generate_fixtures.rake]: or maybe these should be exact matches? you have a nice error message if they're not. then we don't downcase anything.

File: lib/tasks/generate_fixtures.rake [hunk 6]
```diff
@@ -1,64 +1,109 @@
...
-      measure_file = args[:gen_test] ? File.join(fixtures_directory, "draft_measures", args[:test_set], test_name, "measures.json") : File.join(output_directory, "measure.json")
-      value_sets_dir = args[:gen_test] ? File.join(fixtures_directory, "health_data_standards_svs_value_sets", args[:test_set], test_name) : File.join(output_directory, "value_sets")
+    def create_fixture_file(file_path, fixture_json)
+      FileUtils.mkdir_p(File.dirname(file_path)) unless Dir.exists? File.dirname(file_path)
+      File.new(file_path, "w+")
+      File.write(file_path, fixture_json)
+    end
 
+    ###
+    # cms_hqmf: Use cms or hqmf id.
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: Change comment to "indicates if a CMS id or an HQMF id is used"

File: lib/tasks/generate_fixtures.rake [hunk 7]
```diff
@@ -1,64 +1,109 @@
...
+    #   values: cms, hqmf
+    # path: Fixture path.
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: Change to "path to output fixture files" or something similarly more explanatory

File: lib/tasks/generate_fixtures.rake [hunk 8]
```diff
@@ -1,64 +1,109 @@
...
+    # test_name: Name of the test the fixture is for.
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: What about `test_type`? That makes this seem more generalizable. I'd like to use the same set of fixtures across multiple tests as long as I want similar sort of behavior (e.g. proportion measure with multiple populations).

File: lib/tasks/generate_fixtures.rake [hunk 9]
```diff
@@ -1,64 +1,109 @@
...
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: should we mention that the measure needs to be contained in user_email's account?

File: lib/tasks/generate_fixtures.rake [hunk 10]
```diff
@@ -1,64 +1,109 @@
...
+      File.new(file_path, "w+")
+      File.write(file_path, fixture_json)
+    end
 
+    ###
+    # cms_hqmf: Use cms or hqmf id.
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # test_name: Name of the test the fixture is for.
+    # user_email: email of user to export
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: should we also add something about this being the account that will be used for patients/measure/etc.?

File: lib/tasks/generate_fixtures.rake [hunk 11]
```diff
@@ -1,64 +1,109 @@
...
-      patient_file = args[:gen_test] ? File.join(fixtures_directory, "records", args[:test_set], test_name, "patient.json") : File.join(output_directory, "patient.json")
-      measure_file = args[:gen_test] ? File.join(fixtures_directory, "draft_measures", args[:test_set], test_name, "measures.json") : File.join(output_directory, "measure.json")
-      value_sets_dir = args[:gen_test] ? File.join(fixtures_directory, "health_data_standards_svs_value_sets", args[:test_set], test_name) : File.join(output_directory, "value_sets")
+    def create_fixture_file(file_path, fixture_json)
+      FileUtils.mkdir_p(File.dirname(file_path)) unless Dir.exists? File.dirname(file_path)
+      File.new(file_path, "w+")
+      File.write(file_path, fixture_json)
+    end
 
+    ###
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: Can you confirm that this comment follows ruby comment guidance? Talk with Hoss for where to find the guidance. Most of our comments don't follow this, but since this is so comprehensive and for an important rake task, I think it would be good to follow the guidance here.

File: lib/tasks/generate_fixtures.rake [hunk 12]
```diff
@@ -1,64 +1,109 @@
...
+    # cms_hqmf: Use cms or hqmf id.
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # test_name: Name of the test the fixture is for.
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
+    # bundle exec rake HDS:test:generate_fixtures[cms,test/CMSFakevFake,initialLoad,fake@fake,CMSFakevFake]
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: Can you note that this is an example usage?
Can you use real-ish data? the "FakevFake" is hard to parse.
To ensure there are no typos, it would be good to use an example that you were actually able to execute on your machine.

Comment (edeyoung) [reply lib/tasks/generate_fixtures.rake]: Also, please add spaces after the commas for readability.

File: lib/tasks/generate_fixtures.rake [hunk 13]
```diff
@@ -1,64 +1,109 @@
...
+    ###
+    task :generate_frontend_fixture, [:cms_hqmf, :path, :test_name, :user_email, :measure_id] => [:environment] do |t, args|
+      user = User.find_by email: args[:user_email]
+      measure = get_measure(user, args[:cms_hqmf], args[:measure_id])
+      
+      cms_dir = File.join(args[:test_name], args[:measure_id])
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: why is this called `cms_dir`? Should this be `output_dir` or something similar?

Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: for `test_name` may be worth adding to the comment above that this will be used to build the path to the tests along with the `path` argument.

File: lib/tasks/generate_fixtures.rake [hunk 14]
```diff
@@ -1,64 +1,109 @@
...
+      
+      measure_summaries = UploadSummary::MeasureSummary.by_user_and_hqmf_set_id(user, measure.hqmf_set_id).first
+      archived_measures = ArchivedMeasure.by_user_and_hqmf_set_id(user, measure.hqmf_set_id)
+      
+      fixtures_path = File.join('spec', 'javascripts', 'fixtures', 'json')
+      
+      measure_file = File.join(fixtures_path, 'measure_data', args[:path], cms_dir, 'measures.json')
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: add a comment above this along the lines of "generating the fixtures for the measure. these go in a `measure_data` parent directory. the measure file is called `measures.json`. The accompanying value sets file is called `value_sets.json`"

File: lib/tasks/generate_fixtures.rake [hunk 15]
```diff
@@ -1,64 +1,109 @@
...
-        FileUtils.mkdir_p(File.dirname(patient_file)) unless Dir.exists? File.dirname(patient_file)
-        FileUtils.mkdir_p(File.dirname(measure_file)) unless Dir.exists? File.dirname(measure_file)
-        FileUtils.mkdir_p(value_sets_dir) unless Dir.exists? value_sets_dir
+      records = Record.by_user_and_hqmf_set_id(user, measure.hqmf_set_id)
+      record_file = File.join(fixtures_path, 'records', args[:path], cms_dir, "patients.json")
+      create_fixture_file(record_file, JSON.pretty_generate(JSON.parse(records.to_json)))
+      
+      oid_to_vs_map = {}
+      
+      value_sets = measure.value_sets.each do |vs|
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: move this value set stuff up to the measure stuff to create cohesion in where things are being generated.

File: lib/tasks/generate_fixtures.rake [hunk 16]
```diff
@@ -1,64 +1,109 @@
...
+      
+      measure_file = File.join(fixtures_path, 'measure_data', args[:path], cms_dir, 'measures.json')
+      create_fixture_file(measure_file, JSON.pretty_generate(JSON.parse([measure].to_json)))
 
-      if args[:gen_test]
-        FileUtils.mkdir_p(File.dirname(user_file)) unless Dir.exists? File.dirname(user_file)
-        FileUtils.mkdir_p(File.dirname(patient_file)) unless Dir.exists? File.dirname(patient_file)
-        FileUtils.mkdir_p(File.dirname(measure_file)) unless Dir.exists? File.dirname(measure_file)
-        FileUtils.mkdir_p(value_sets_dir) unless Dir.exists? value_sets_dir
+      records = Record.by_user_and_hqmf_set_id(user, measure.hqmf_set_id)
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: isn't the fixture for a record a bit different on the front end than the back end?

Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: add a similar high level comment as described above to this patient section.

File: lib/tasks/generate_fixtures.rake [hunk 17]
```diff
@@ -1,64 +1,109 @@
...
+      
+      oid_to_vs_map = {}
+      
+      value_sets = measure.value_sets.each do |vs|
+        oid_to_vs_map[vs.oid] = vs
       end
+      value_sets_file = File.join(fixtures_path, 'measure_data', args[:path], cms_dir, 'value_sets.json')
+      create_fixture_file(value_sets_file, JSON.pretty_generate(JSON.parse(oid_to_vs_map.to_json)))
+    
+      us_filename = "#{args[:test_name]}.json"
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: rename to upload_summaries_filename or something similar. `us` is confusing.

File: lib/tasks/generate_fixtures.rake [hunk 18]
```diff
@@ -1,64 +1,109 @@
...
+    # measure_id: id of measuer to export
+    # bundle exec rake HDS:test:generate_fixtures[cms,test/CMSFakevFake,initialLoad,fake@fake,CMSFakevFake]
+    ###
+    task :generate_frontend_fixture, [:cms_hqmf, :path, :test_name, :user_email, :measure_id] => [:environment] do |t, args|
+      user = User.find_by email: args[:user_email]
+      measure = get_measure(user, args[:cms_hqmf], args[:measure_id])
+      
+      cms_dir = File.join(args[:test_name], args[:measure_id])
+      
+      measure_summaries = UploadSummary::MeasureSummary.by_user_and_hqmf_set_id(user, measure.hqmf_set_id).first
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: consider moving these declarations down to where the fixtures get generated for them.

File: lib/tasks/generate_fixtures.rake [hunk 19]
```diff
@@ -1,64 +1,109 @@
...
 
+    ###
+    # cms_hqmf: Use cms or hqmf id.
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # test_name: Name of the test the fixture is for.
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
+    # bundle exec rake HDS:test:generate_fixtures[cms,test/CMSFakevFake,initialLoad,fake@fake,CMSFakevFake]
+    ###
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: consider adding to the comments a comment outlining what the file structure will look like after this is run.

Comment (edeyoung) [reply lib/tasks/generate_fixtures.rake]: also consider adding an explanation on where each of these files should go in the actual spec file structure.

what happened to making these be in one collective space so you wouldn't have to move individual files?

File: lib/tasks/generate_fixtures.rake [hunk 20]
```diff
@@ -1,64 +1,109 @@
...
-      value_sets = measure.value_sets
+    #Usage
+    # cms_hqmf: Use cms or hqmf id.
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
+    # bundle exec rake HDS:test:generate_fixtures[cms,test/CMSFakevFake,fake@fake,CMSFakevFake]
+    desc "Exports a set of fixtures that can be loaded for testing purposes"
+    task :generate_fixtures, [:cms_hqmf, :path, :user_email, :measure_id] => [:environment] do |t, args|
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: should this be renamed to "generate_backend_fixtures"?

File: lib/tasks/generate_fixtures.rake [hunk 21]
```diff
@@ -1,64 +1,109 @@
...
-      File.new(measure_file, "w+")
-      File.write(measure_file, JSON.pretty_generate(JSON.parse(measure.to_json(except: [:map_fns, :record_ids], methods: [:value_sets]))))
-
-      value_sets = measure.value_sets
+    #Usage
+    # cms_hqmf: Use cms or hqmf id.
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: see all comments above re making comments more explanatory.

File: lib/tasks/generate_fixtures.rake [hunk 22]
```diff
@@ -1,64 +1,109 @@
...
+      measure_file = File.join(fixtures_path, 'draft_measures', args[:path], measure_name)
+      create_fixture_file(measure_file, JSON.pretty_generate(JSON.parse(measure.to_json)))
 
-      value_sets.each do |vs|
-        vs_file_name = vs.oid + ".json"
-        value_sets_file = File.join(value_sets_dir, vs_file_name)
-        File.new(value_sets_file, "w+")
-        File.write(value_sets_file, JSON.pretty_generate(JSON.parse(vs.to_json)))
+      records = Record.by_user_and_hqmf_set_id(user, measure.hqmf_set_id).each do |rec|
+        rec_name = rec.first + "_" + rec.last + ".json"
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: the first and last names often have spaces in them. does it make sense (as painful as it is) to use a space here instead of an underscore?

File: lib/tasks/generate_fixtures.rake [hunk 23]
```diff
@@ -1,64 +1,109 @@
...
-
-      value_sets = measure.value_sets
+    #Usage
+    # cms_hqmf: Use cms or hqmf id.
+    #   values: cms, hqmf
+    # path: Fixture path.
+    # user_email: email of user to export
+    # measure_id: id of measuer to export
+    # bundle exec rake HDS:test:generate_fixtures[cms,test/CMSFakevFake,fake@fake,CMSFakevFake]
+    desc "Exports a set of fixtures that can be loaded for testing purposes"
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: it seems like there is some repeated code in here. Is it worth refactoring to reduce repeated code?

**Review by edeyoung — CHANGES_REQUESTED** [review]
Please excuse my brevity. rushed the comments. i'm curious about the resulting file structure. let's discuss. i'm also curious because the fixtures for front and back end seem the same other than being in an array or separate. i thought there were differences. does it turn out there aren't?

File: lib/tasks/generate_fixtures.rake [hunk 24]
```diff
@@ -1,83 +1,127 @@
...
+    def set_mongoid_ids(json)
+      if json.kind_of?( Hash)
+        json.each_pair do |k,v|
+          if v && v.kind_of?( Hash )
+            if v["$oid"]
+              json[k] = BSON::ObjectId.from_string(v["$oid"])
+            else
+              set_mongoid_ids(v)
+            end
+          elsif k == '_id' || k == 'bundle_id' || k == 'user_id'
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: does this cause an issue with `cms_id`?

File: lib/tasks/generate_fixtures.rake [hunk 25]
```diff
@@ -1,83 +1,127 @@
...
+            puts "Converting #{k} : value #{v}"
+            json[k] = BSON::ObjectId.from_string(v)   
+            puts "Converted: #{k} to #{json[k]}"   
+          end
+        end
+      end
+    end
+    
+    def collection_fixtures(*collection_names)
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: are these duplicated from elsewhere in the code? if so, can these just call that code? if not, please add a comment explaining why the duplication was necessary.

File: lib/tasks/generate_fixtures.rake [hunk 26]
```diff
@@ -1,83 +1,127 @@
...
-    # cms_hqmf: Use cms or hqmf id.
+    # cms_hqmf: indicates if a CMS id or an HQMF id is used.
     #   values: cms, hqmf
-    # path: Fixture path.
-    # test_name: Name of the test the fixture is for.
+    # path: Path to fixture files, derived from the data-type directories (EX: measure_data/${path}).
     # user_email: email of user to export
     # measure_id: id of measuer to export
-    # bundle exec rake HDS:test:generate_fixtures[cms,test/CMSFakevFake,initialLoad,fake@fake,CMSFakevFake]
+    # bundle exec rake HDS:test:generate_frontend_fixtures[cms,test/CMSFakevFake,initialLoad,fake@fake,CMSFakevFake]
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: `HDS:test:` got changed to `bonnie:fixtures` above

Comment (edeyoung) [reply lib/tasks/generate_fixtures.rake]: try using real data for the example. the fake data is hard to read.

File: lib/tasks/generate_fixtures.rake [hunk 27]
```diff
@@ -1,83 +1,127 @@
...
     end
 
     ###
-    # cms_hqmf: Use cms or hqmf id.
+    # cms_hqmf: indicates if a CMS id or an HQMF id is used.
     #   values: cms, hqmf
-    # path: Fixture path.
-    # test_name: Name of the test the fixture is for.
+    # path: Path to fixture files, derived from the data-type directories (EX: measure_data/${path}).
     # user_email: email of user to export
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: from previous review: "should we also add something about this being the account that will be used for patients/measure/etc.?"

File: lib/tasks/generate_fixtures.rake [hunk 28]
```diff
@@ -1,83 +1,127 @@
...
     # measure_id: id of measuer to export
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: from previous review: "should we mention that the measure needs to be contained in user_email's account?
 Show outdated"

File: lib/tasks/generate_fixtures.rake (hunk 26 above)
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: from the previous review: 
Can you note that this is an example usage?
Can you use real-ish data? the "FakevFake" is hard to parse.
To ensure there are no typos, it would be good to use an example that you were actually able to execute on your machine.

File: lib/tasks/generate_fixtures.rake [hunk 29]
```diff
@@ -1,83 +1,127 @@
...
-    task :generate_frontend_fixture, [:cms_hqmf, :path, :test_name, :user_email, :measure_id] => [:environment] do |t, args|
+    task :generate_frontend_fixtures, [:cms_hqmf, :path, :user_email, :measure_id] => [:environment] do |t, args|
       user = User.find_by email: args[:user_email]
       measure = get_measure(user, args[:cms_hqmf], args[:measure_id])
       
-      cms_dir = File.join(args[:test_name], args[:measure_id])
       
-      measure_summaries = UploadSummary::MeasureSummary.by_user_and_hqmf_set_id(user, measure.hqmf_set_id).first
+      measure_summaries = UploadSummary::MeasureSummary.by_user_and_hqmf_set_id(user, measure.hqmf_set_id).desc(:created_at)
       archived_measures = ArchivedMeasure.by_user_and_hqmf_set_id(user, measure.hqmf_set_id)
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: from previous review: consider moving these declarations down to where the fixtures get generated for them.
 Hide outdated

File: lib/tasks/generate_fixtures.rake [hunk 30]
```diff
@@ -1,83 +1,127 @@
...
       
       fixtures_path = File.join('spec', 'javascripts', 'fixtures', 'json')
       
-      measure_file = File.join(fixtures_path, 'measure_data', args[:path], cms_dir, 'measures.json')
+      measure_file = File.join(fixtures_path, 'measure_data', args[:path], 'measures.json')
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: from previous review: add a comment above this along the lines of "generating the fixtures for the measure. these go in a measure_data parent directory. the measure file is called measures.json. The accompanying value sets file is called value_sets.json"

File: lib/tasks/generate_fixtures.rake [hunk 31]
```diff
@@ -1,83 +1,127 @@
...
-      create_fixture_file(record_file, JSON.pretty_generate(JSON.parse(records.to_json)))
-      
       oid_to_vs_map = {}
       
       value_sets = measure.value_sets.each do |vs|
         oid_to_vs_map[vs.oid] = vs
       end
-      value_sets_file = File.join(fixtures_path, 'measure_data', args[:path], cms_dir, 'value_sets.json')
+
+      records = Record.by_user_and_hqmf_set_id(user, measure.hqmf_set_id)
```
Comment (edeyoung) [inline lib/tasks/generate_fixtures.rake]: from previous review: add a similar high level comment as described above to this patient section.
 Show outdated

**Review by edeyoung — CHANGES_REQUESTED** [review]
Hey, it looks like several of my comments were missed. github is doing weird stuff which is making  me wonder if maybe they didn't show up. If they did show up and you disagreed with them so didn't implement them, can you make sure to respond to the comment noting that? Thanks!

File: lib/tasks/generate_fixtures.rake (hunk 24 above)
Comment (alexanderelliott121) [reply lib/tasks/generate_fixtures.rake]: Nope

Comment (edeyoung) [reply lib/tasks/generate_fixtures.rake]: why not?

Comment (alexanderelliott121) [reply lib/tasks/generate_fixtures.rake]: k == '_id' isn't a regex, so it doesn't recognize/convert the cms_id.
cms_id isn't a bson object, so no conversion is required.

**Comment by edeyoung** [comment]
@alexanderelliott121 

I do not seem to be able to login to the bonnie user after I wipe my db.

please check out my very minor changes.

**Review by edeyoung — APPROVED** [review]
